### PR TITLE
Add api proxy

### PR DESCRIPTION
**Pull request by cheyang**
### Ⅰ. Describe what this PR does

Add API proxy for users to run API client: https://docs.alluxio.io/os/user/stable/en/api/FS-API.html#python

### Ⅴ. Special notes for reviews

### CODE REVIEW

**Review by xieydd — APPROVED** [review]
/LGTM

File: pkg/ddc/alluxio/transform_api_gateway.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import (
+	"strconv"
+
+	datav1alpha1 "github.com/fluid-cloudnative/fluid/api/v1alpha1"
+)
+
+// transformAPIGateway transforms the given value
+func (e *AlluxioEngine) transformAPIGateway(runtime *datav1alpha1.AlluxioRuntime, value *Alluxio) (err error) {
+
+	if runtime.Spec.APIGateway.Enabled {
```
Comment (TrafalgarZZZ) [inline pkg/ddc/alluxio/transform_api_gateway.go:12]: Is it a redundancy here?  It seems the property is already set in `setPortProperties` func

File: pkg/ddc/alluxio/transform.go [hunk 2]
```diff
@@ -350,21 +357,55 @@ func (e *AlluxioEngine) transformWorkers(runtime *datav1alpha1.AlluxioRuntime, v
...
-		value.JobMaster.Ports.Rpc = allocatedPorts[4]
-		value.JobMaster.Ports.Web = allocatedPorts[5]
-		value.JobWorker.Ports.Rpc = allocatedPorts[6]
-		value.JobWorker.Ports.Web = allocatedPorts[7]
-		value.JobWorker.Ports.Data = allocatedPorts[8]
-	} else {
-		value.Master.Ports.Embedded = allocatedPorts[9]
-		value.JobMaster.Ports.Embedded = allocatedPorts[10]
+	// check the length ports
+	if len(allocatedPorts) < PORT_NUM {
```
[the remark this answers is not in the document]
Comment (cheyang) [reply pkg/ddc/alluxio/transform.go:361]: Yes， I have opened the an feature on it.

File: pkg/ddc/alluxio/transform_api_gateway.go (hunk 1 above)
Comment (cheyang) [reply pkg/ddc/alluxio/transform_api_gateway.go:12]: Yes, you are correct.

**Pull request merged**